File: fabric/src/main/java/com/cobblemontournament/fabric/CobblemonTournamentFabric.java
package com.cobblemontournament.fabric;

import com.cobblemontournament.common.TournamentModImplementation;
import com.cobblemontournament.common.config.ConfigProvider;
import com.cobblemontournament.common.CobblemonTournament;
import com.cobblemontournament.common.commands.TournamentCommands;
import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback;
import net.fabricmc.loader.api.FabricLoader;

public class CobblemonTournamentFabric implements ModInitializer, TournamentModImplementation
{
    @Override
    public void onInitialize() {
        CobblemonTournament.initialize (this);
    }

    @Override
    public void initializeConfig()
    {
        ConfigProvider.registerConfigs(FabricLoader.getInstance().getConfigDir());
    }

    @Override
    public void registerEvents()
    {
        CobblemonTournament.registerEvents();
    }

    @Override
    public void registerCommands()
    {
        CommandRegistrationCallback.EVENT.register(TournamentCommands::register);
    }
    
}

File: fabric/src/main/kotlin/com/cobblemontournament/fabric/CobblemonTournamentFabric.kt
package com.cobblemontournament.fabric

import com.cobblemontournament.common.CobblemonTournament
import com.cobblemontournament.common.config.TournamentConfig
import com.someguy.mod.PlatformModImplementation
import net.fabricmc.api.ModInitializer

class CobblemonTournamentFabric :
    PlatformModImplementation(common = CobblemonTournament),
    ModInitializer {

    override fun onInitialize() = initialize()

    // no platform specific initialization needed at this time
    override fun initialize() = initializeCommon()

    override fun initializeConfig() = TournamentConfig.registerConfigs()

    // no platform specific events to register at this time
    override fun registerEvents() { }

}
